node/impl/client/client.go

```diff
@@ -85,12 +85,16 @@ func (a *API) ClientStartDeal(ctx context.Context, params *api.StartDealParams)
 		return nil, xerrors.Errorf("failed checking miners sector size: %w", err)
 	}
 
+	if uint64(params.Data.PieceSize) > uint64(ssize) {
```

Comment:
This needs to be
```suggestion
	if uint64(params.Data.PieceSize.Padded()) > uint64(ssize) {
```

Otherwise you'd allow pieces that are 1.00787401575 the sector size

Reply:
Done, thanks for the catch :)